src/sd_card_signer.rs

```diff
@@ -0,0 +1,79 @@
+use anyhow::Context;
+use bdk::{
+    bitcoin::{
+        secp256k1::{All, Secp256k1},
+        util::psbt::PartiallySignedTransaction,
+    },
+    wallet::signer::{Signer, SignerError, SignerId},
+};
+use core::str::FromStr;
+use std::path::PathBuf;
+
+#[derive(Debug)]
+pub struct SDCardSigner {
+    psbt_output_dir: PathBuf,
+}
+
+impl SDCardSigner {
+    pub fn create(psbt_output_dir: PathBuf) -> Self {
+        SDCardSigner { psbt_output_dir }
+    }
+}
+
+impl Signer for SDCardSigner {
+    fn sign(
+        &self,
+        psbt: &mut PartiallySignedTransaction,
+        _input_index: Option<usize>,
+        _secp: &Secp256k1<All>,
+    ) -> Result<(), SignerError> {
+        // Probably should figure out how to incorporate SignerErrors here
+        let txid = psbt.clone().extract_tx().txid();
+        let mut psbt_file = PathBuf::from(self.psbt_output_dir.clone());
+        psbt_file.push(format!("{}.psbt", txid.to_string()));
+
+        let _ = std::fs::write(psbt_file.clone(), psbt.to_string())
+            .context("writing PSBT file to SD path");
```

Comment:
This is doing nothing since you didn't propagate the error. I think you have to eprintln these errors.